### ISSUE TITLE
@ symbol in JSON keys produces private variable 

### ISSUE THREAD

**Issue by glenrobson** [koxudaxi/datamodel-code-generator#917]
**Describe the bug**
We have JSON that has `@id` and `@type` which are reflected in the schema below. When we generate the python model using the datamodel-codegen library it produces variables that we can't access. This is likely due to [pydantic hiding variables](https://pydantic-docs.helpmanual.io/usage/models/#automatically-excluded-attributes) starting with an underscore. 

**To Reproduce**

Example schema:
```json
{
    "$id": "schema_v2.json",
    "$schema": "http://json-schema.org/schema#",
    
    "type": "object",
    "properties": {
        "@id": { 
            "type": "string",
            "format": "uri",
            "pattern": "^http.*$",
            "title": "Id must be presesnt and must be a URI"
        },    
        "@type": { "type": "string" },
        "profile": { "type": "string" }
    },
    "required": ["@id", "@type"]
} 
```

Produces the following model:
```
class Model(BaseModel):
    _id: AnyUrl = Field(..., alias='@id', title='Id must be presesnt and must be a URI')
    _type: str = Field(..., alias='@type')
    profile: Optional[str] = None
```

and if we try and use it we can't access id or type:
```
$ python
Python 3.9.10 (main, Jan 15 2022, 11:48:04) 

>>> from model2 import Model
>>> v2 = Model(_id="http://example.com/iiif/2", _type="service")
>>> print (v2)
profile=None
```

Used commandline:
```
$ datamodel-codegen  --input schema_v2.json --input-file-type jsonschema --output model2.py
```

**Expected behavior**
Ideally we would look the following conversion where the _ is removed:

```
class Model(BaseModel):
    id: AnyUrl = Field(..., alias='@id', title='Id must be presesnt and must be a URI')
    type: str = Field(..., alias='@type')
    profile: Optional[str] = None
```

**Version:**
 - OS:  Mac OS X Monterey 12.6
 - Python version:  3.9.10
 - datamodel-code-generator version: 0.13.5

**Additional context**
JSON keys `@id`, `@type` and `@context` are used in [JSON-LD](https://json-ld.org/)

**Comment by tobiasfeil** [comment]
I have a differen use case, but the same problem surfaces.

I'm consuming an API where one of the models has a field that starts with `_`. The field makes it into the generated code, but Pydantic ignores the field upon model instantiation, thinking I want it to be private.

It would be great if there was a CLI flag that makes it so the appropriate option is included in the generated models where I can use the `_` fields just like any other fields.

**Comment by tobiasfeil** [comment]
Man, you're an ABSOLUTE LEGEND. Such fast response time, especially over the holidays. You've earned yourself a coffee :money_mouth_face:

**Comment by koxudaxi** [comment]
@tobiasfeil 
Thank you for your supporting me :smile: 
This month,  I have come back to OSS world. and I resolve a bunch of stuck issues every night.
It means you are very lucky:)

**Comment by tobiasfeil** [comment]
Great for me! Welcome back :)

**Comment by koxudaxi** [comment]
@glenrobson @tobiasfeil 
I have released the new version as 0.14.1 :tada:
